### PR TITLE
Group all privileged logic together

### PR DESCRIPTION
**Pull request by miaoyq**
Related to https://github.com/kubernetes-incubator/cri-containerd/blob/dcc3cb2a05400b4dbc565d153127f23c06c823a1/pkg/server/container_create.go#L195
/cc @Random-Liu 
Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

File: pkg/server/container_create.go [hunk 1]
```diff
@@ -298,6 +303,17 @@ func addImageEnvs(g *generate.Generator, imageEnvs []string) error {
...
 }
 
+func setOCIPrivileged(g *generate.Generator, config *runtime.ContainerConfig) error {
+	// Add all capabilities in privileged mode.
+	g.SetupPrivileged(true)
+	setOCIBindMountsPrivilege(g)
+	if err := addOCIDevicesWithPrivilege(g); err != nil {
+		return fmt.Errorf("failed to set devices mapping %+v: %v", config.GetDevices(), err)
+	}
+
```
Comment (Random-Liu) [inline pkg/server/container_create.go]: unnecessary empty line.

File: pkg/server/container_create.go [hunk 2]
```diff
@@ -192,44 +192,49 @@ func (c *criContainerdService) generateContainerSpec(id string, sandboxPid uint3
...
-	addOCIBindMounts(&g, append(extraMounts, config.GetMounts()...), securityContext.GetPrivileged())
+	addOCIBindMounts(&g, append(extraMounts, config.GetMounts()...))
 
-	g.SetRootReadonly(securityContext.GetReadonlyRootfs())
+	if securityContext.GetPrivileged() {
+		if err := setOCIPrivileged(&g, config); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := addOCIDevicesWithoutPrivilege(&g, config.GetDevices()); err != nil {
```
Comment (Random-Liu) [inline pkg/server/container_create.go]: Please follow the same pattern. For the other functions, you are using `setOCICapabilities` vs SetupPrivileged, `addOCIBindMounts` vs `setOCIBindMountsPrivileged`.

How about we use `addOCIDevices` vs `setOCIDevicesPrivileged` to make them consistent?

Comment (miaoyq) [inline pkg/server/container_create.go]: @Random-Liu Yes, it's nice, Thanks!
Done!

File: pkg/server/container_create.go (hunk 1 above)
Comment (miaoyq) [inline pkg/server/container_create.go]: Done!

File: pkg/server/container_create.go (hunk 2 above)
Comment (Random-Liu) [inline pkg/server/container_create.go]: s/addOCIDevicesPrivileged/setOCIDevicesPrivileged.

**Comment by Random-Liu** [comment]
@miaoyq One more comment. Otherwise LGTM.

**Comment by miaoyq** [comment]
@Random-Liu  I'm sorry for my careless, Done!

**Comment by Random-Liu** [comment]
@miaoyq LGTM. Thanks for addressing this.

Will merge after test passes.

**Pull request merged**